matsim/src/main/java/org/matsim/core/mobsim/qsim/QSim.java

```diff
@@ -252,7 +252,12 @@ public void run() {
 			// We really want to perform that. For instance, with QNetsimEngine, threads are cleaned up in this method.
 			// Without this finally, in case of a crash, threads are not closed, which lead to process hanging forever
 			// at least on the eth euler cluster (but not on our local machines at ivt!?) td oct 15
-			cleanupSim();
+			try {
+				cleanupSim();
+			} catch(Exception e) {
+				log.warn("exception in finally block - this may be a follow-up exception of an exception thrown in the try block.");

```

Comment:
Instead of using a separate `e.printStackTrace()`, I would add the exception to the logging statement, so it all comes together as one: 
```suggestion
				log.warn("exception in finally block - this may be a follow-up exception of an exception thrown in the try block.", e);
```
Then, one could get rid of the line below.

Reply:
good idea!